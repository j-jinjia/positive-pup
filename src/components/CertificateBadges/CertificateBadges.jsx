import "./CertificateBadges.scss";
import internationalBadge from "../../assets/images/certification_badges/international_companion.png";
import ppgBadge from "../../assets/images/certification_badges/PPG_Member_Badge.png";
import ukBehaviour from "../../assets/images/certification_badges/UK_dog_behaviour.png";
import victoriaStillwell from "../../assets/images/certification_badges/victoria_stillwell.png";


const CertificateBadges = (props) => {

  const {showText} = props;
  let certificateClassName = "certification-section";
  certificateClassName += showText ? " certification-section--show" : " certification-section--hide";


  return (
    <>
      <div className={certificateClassName}>
        <h2 className="certification-section__header">You’re in professional hands!</h2>
        <p className="certification-section__paragraph">I’m a well-established, certified trainer in Dog behaviour &#38; training, withover  <span className="certification-section__paragraph--highlighted-text">4.5 years of experience.</span> To find out moreinformation  about any of my certifications, please click on one of the logos.</p>
      </div>    

      <div className="certificate-badges">
        <a
          href="https://www.vsdogtrainingacademy.com/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            src={victoriaStillwell}
            className="certificate-badges__img"
            alt="Victoria Stillwell certification badge"
          />
        </a>
        <a
          href="https://companionanimal.network/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            src={internationalBadge}
            className="certificate-badges__img"
            alt="international companion animal network certification badge"
          />
        </a>
        <a href="https://ukdogcharter.org/" rel="noreferrer" target="_blank">
          <img
            src={ukBehaviour}
            className="certificate-badges__img"
            alt="UK dog behaviour and training charter certification badge"
          />
        </a>
        <a
          href="https://www.petprofessionalguild.com/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            src={ppgBadge}
            className="certificate-badges__img"
            alt="the pet professional guild certification badge"
          />
        </a>
      </div>
    </>
  );
};

export default CertificateBadges;
