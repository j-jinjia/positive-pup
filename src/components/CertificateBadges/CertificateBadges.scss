@use "../../styles/variables" as *;

.certification-section--show,
.certification-section {
  display: flex;
  align-items: center;
  flex-direction: column;
  padding: 2rem;

  &__header {
    font-size: $font-size-h2;
    margin-bottom: 2rem;
    line-height: 1.1;
  }

  &__paragraph {
    margin-bottom: 2rem;
    line-height: 2;
    font-size: 1.25rem;
    color: $colour-tertiary;

    &--highlighted-text {
      color: $colour-primary;
      font-weight: 700;
      text-decoration: underline;
      text-underline-offset: 0.2rem;
    }
  }
}

.certification-section--hide {
  display: none;
}

.certificate-badges {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  &__img {
    margin: 0px 10px;
    width: 60px;
    height: 60px;
  }
}

@media screen and (min-width: 460px) {
  .certificate-badges {
    justify-content: center;
    width: 450px;

    &__img {
      margin: 0px 19px;
      width: auto;
      height: auto;
    }
  }
}

@media screen and (min-width: $tablet-breakpoint) {
  .certification-section {
    text-align: center;
  }
}

@media screen and (min-width: $desktop-breakpoint) {
  .certification-section {
    max-width: 700px;
  }
}
