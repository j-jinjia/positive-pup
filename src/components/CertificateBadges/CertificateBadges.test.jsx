import { customRender } from "../../utils/testUtils";
import CertificateBadges from "./CertificateBadges";

it("should render the home page", () => {
  const { container } = customRender(<CertificateBadges />);

  expect(container).toMatchSnapshot();
});
