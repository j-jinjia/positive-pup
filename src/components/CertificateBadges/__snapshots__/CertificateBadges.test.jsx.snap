// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the home page 1`] = `
<div>
  <div
    class="certification-section certification-section--hide"
  >
    <h2
      class="certification-section__header"
    >
      You’re in professional hands!
    </h2>
    <p
      class="certification-section__paragraph"
    >
      I’m a well-established, certified trainer in Dog behaviour & training, withover  
      <span
        class="certification-section__paragraph--highlighted-text"
      >
        4.5 years of experience.
      </span>
       To find out moreinformation  about any of my certifications, please click on one of the logos.
    </p>
  </div>
  <div
    class="certificate-badges"
  >
    <a
      href="https://www.vsdogtrainingacademy.com/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Victoria Stillwell certification badge"
        class="certificate-badges__img"
        src="victoria_stillwell.png"
      />
    </a>
    <a
      href="https://companionanimal.network/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="international companion animal network certification badge"
        class="certificate-badges__img"
        src="international_companion.png"
      />
    </a>
    <a
      href="https://ukdogcharter.org/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="UK dog behaviour and training charter certification badge"
        class="certificate-badges__img"
        src="UK_dog_behaviour.png"
      />
    </a>
    <a
      href="https://www.petprofessionalguild.com/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="the pet professional guild certification badge"
        class="certificate-badges__img"
        src="PPG_Member_Badge.png"
      />
    </a>
  </div>
</div>
`;
